### ISSUE TITLE
Daemon crashed with Sent_invalid_signature_or_proof("Signature failed to verify")

### ISSUE THREAD

**Issue by dheeraj07** [MinaProtocol/mina#7560]
Please find the attached logs

<img width="1028" alt="Screen Shot 2021-01-21 at 12 43 55 AM" src="https://user-images.githubusercontent.com/22843503/105225147-2b961980-5b84-11eb-8f86-4530db60fe68.png">

https://drive.google.com/file/d/1Muq0FJVMjmDGnnYXcnzh6dsrggeSSZ1F/view?usp=sharing

**Comment by mrmr1993** [comment]
The last several log lines are:
```json
{
  "timestamp": "2021-01-20 18:16:21.329125Z",
  "level": "Faulty_peer",
  "source": {
    "module": "Trust_system__Peer_trust",
    "location": "File \"src/lib/trust_system/peer_trust.ml\", line 185, characters 10-51"
  },
  "message": "Banning peer $sender_id until $expiration due to $action",
  "metadata": {
    "action": "Trust_system.Actions.Sent_invalid_signature_or_proof (Staged_ledger error: $error)",
    "error": "Pre_diff_info.Error error: (Verification_failed (Verification_failed \"signature failed to verify\"))",
    "expiration": "2021-01-21 18:16:21.329110Z",
    "host": "135.181.176.189",
    "peer_id": "12D3KooWEmAo5xheX8r5ZJnNAYrtqXn5XfZxsV6U9kFo5PFeRXdZ",
    "pid": 12,
    "port": 8302,
    "sender_id": {
      "host": "195.201.175.114",
      "peer_id": "12D3KooWKRb1WyoUX9nAyrmqcnWBUZAiAU1BNXTPbwYjdmgpvMjf",
      "libp2p_port": 8302
    }
  },
  "event_id": "4d3d3e37e0365f1dbc6d35fddafeb77d"
}
{
  "timestamp": "2021-01-20 18:16:21.791501Z",
  "level": "Error",
  "source": {
    "module": "Cache_lib__Impl",
    "location": "File \"src/lib/cache_lib/impl.ml\", line 222, characters 10-48"
  },
  "message": "Cached.sequence_result called on an already consumed Cached.t",
  "metadata": {
    "cache": "Transition_handler__Unprocessed_transition_cache",
    "host": "135.181.176.189",
    "peer_id": "12D3KooWEmAo5xheX8r5ZJnNAYrtqXn5XfZxsV6U9kFo5PFeRXdZ",
    "pid": 12,
    "port": 8302
  }
}
{
  "timestamp": "2021-01-20 18:16:21.791640Z",
  "level": "Error",
  "source": {
    "module": "Transition_handler__Processor",
    "location": "File \"src/lib/transition_handler/processor.ml\", line 190, characters 14-26"
  },
  "message": "Error while building breadcrumb in the transition handler processor: $error",
  "metadata": {
    "error": {
      "string": "Pre_diff_info.Error error: (Verification_failed (Verification_failed \"signature failed to verify\"))"
    },
    "host": "135.181.176.189",
    "peer_id": "12D3KooWEmAo5xheX8r5ZJnNAYrtqXn5XfZxsV6U9kFo5PFeRXdZ",
    "pid": 12,
    "port": 8302,
    "state_hash": "3NKMLqDmgc4Mq4QHfzi4QQQJgwSNqXtRRn9xqQdV2saLs7PcErPA"
  }
}
{
  "timestamp": "2021-01-20 18:16:21.791679Z",
  "level": "Debug",
  "source": {
    "module": "Network_pool__Batcher",
    "location": "File \"src/lib/network_pool/batcher.ml\", line 130, characters 4-26"
  },
  "message": "Verifying proofs in batch of size $num_proofs",
  "metadata": {
    "host": "135.181.176.189",
    "num_proofs": 2,
    "peer_id": "12D3KooWEmAo5xheX8r5ZJnNAYrtqXn5XfZxsV6U9kFo5PFeRXdZ",
    "pid": 12,
    "port": 8302
  }
}
{
  "timestamp": "2021-01-20 18:16:21.791688Z",
  "level": "Debug",
  "source": {
    "module": "Network_pool__Batcher",
    "location": "File \"src/lib/network_pool/batcher.ml\", line 153, characters 4-26"
  },
  "message": "Calling verifier with $num_proofs on $ids",
  "metadata": {
    "host": "135.181.176.189",
    "ids": [
      8118,
      8119
    ],
    "num_proofs": 2,
    "peer_id": "12D3KooWEmAo5xheX8r5ZJnNAYrtqXn5XfZxsV6U9kFo5PFeRXdZ",
    "pid": 12,
    "port": 8302
  }
}
{
  "timestamp": "2021-01-20 18:16:22.255401Z",
  "level": "Fatal",
  "source": {
    "module": "Init__Coda_run",
    "location": "File \"src/app/cli/src/init/coda_run.ml\", line 580, characters 2-26"
  },
  "message": "Unhandled top-level exception: $exn\nGenerating crash report",
  "metadata": {
    "exn": {
      "sexp": [
        "monitor.ml.Error",
        "Cached item has already been finalized",
        [
          "Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30",
          "Called from file \"src/lib/cache_lib/impl.ml\", line 173, characters 30-51",
          "Called from file \"src/lib/cache_lib/impl.ml\", line 199, characters 6-69",
          "Called from file \"src/lib/transition_handler/processor.ml\", line 198, characters 16-102",
          "Called from file \"src/deferred0.ml\", line 56, characters 64-69",
          "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5",
          "Called from file \"src/job_queue.ml\", line 171, characters 6-47",
          "Caught by monitor coda"
        ]
      ],
      "backtrace": [
        "Raised by primitive operation at file \"src/jemalloc.ml\", line 53, characters 2-36",
        "Called from file \"src/jemalloc.ml\", line 57, characters 48-67"
      ]
    },
    "host": "135.181.176.189",
    "peer_id": "12D3KooWEmAo5xheX8r5ZJnNAYrtqXn5XfZxsV6U9kFo5PFeRXdZ",
    "pid": 12,
    "port": 8302
  }
}
```

It seems as though sending a staged ledger diff with a bad signature may be triggering misbehaviour in `Cache_lib`, causing a fatal `Cached item has already been finalized` error.

**Comment by mrmr1993** [comment]
This should be fixed by #7568, which will be in the next release. Thanks for reporting!
